crates/bevy_ecs/src/system/input.rs

```diff
@@ -227,3 +224,66 @@ impl<'a, I: SystemInput> SystemInput for StaticSystemInput<'a, I> {
         StaticSystemInput(this)
     }
 }
+
+macro_rules! impl_system_input_tuple {
+    ($(#[$meta:meta])* $($name:ident),*) => {
+        $(#[$meta])*
+        impl<$($name: SystemInput),*> SystemInput for ($($name,)*) {
+            type Param<'i> = ($($name::Param<'i>,)*);
+            type Inner<'i> = ($($name::Inner<'i>,)*);
+
+            #[allow(non_snake_case)]
+            #[allow(clippy::unused_unit)]
+            fn wrap(this: Self::Inner<'_>) -> Self::Param<'_> {
+                let ($($name,)*) = this;
+                ($($name::wrap($name),)*)
+            }
+        }
+    };
+}
+
+all_tuples!(
+    #[doc(fake_variadic)]
+    impl_system_input_tuple,
+    0,
+    8,
```

Comment:
Not a problem with this PR specifically, but I feel like Bevy should probably just have a constant somewhere for `LONGEST_REASONABLE_TUPLE`, instead of each site picking a somewhat random value (I think I've seen 12, 15, and 16, now also 8).

Reply:
That's fair, I just think its way less reasonable to have large-arity input tuples when even using system input at all is somewhat uncommon. So I'd prefer if we could (potentially) save some compilation time by not implementing for arities that are highly unlikely to be used in practice.